crates/biome_service/src/workspace.rs

```diff
@@ -536,8 +538,14 @@ pub struct ProjectFeaturesResult {}
 #[serde(rename_all = "camelCase")]
 pub struct OpenFileParams {
     pub path: BiomePath,
-    pub content: String,
+
+    /// The file's content.
+    ///
+    /// If `None`, the file will opened by the workspace from the local file system.
+    pub content: Option<String>,
```

Comment:
I think having a `Option` isn't exhaustive enough to explain that `content` is either opened by a client or by the workspace. Unless someone reads the comment, a developer could misunderstand that not having some  `content` is a valid use case when it's not.

At this point, I would prefer to have an `enum`, e.g., 

```rs
enum Content {
	FromClient(String),
	FromWorkspace(String)
}
```

or

```rs
enum Content {
	ToAnalyze(String),
	ToInspect(String)
}
```

or `struct`.

Plus, if the `workspace.fs` is going to read the contents regardless (looking at how `open_file`), I don't see the point of sending the content anyways. I believe we should panic if there's no content when calling `open_file`

Reply:
Went for:

```rs
enum FileContent {
	FromClient(String),
	FromWorkspace
}
```